src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,34 +1,79 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
-
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
-
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button
-          data-cy="clearSearchButton"
-          type="button"
-          className="delete"
+import React, { useCallback, useState } from 'react';
+
+import { TodoStatus } from '../../types/TodoStatus';
+import { debounce } from '../../utils/debounce';
+
+type Props = {
+  query: string;
+  setQuery: (query: string) => void;
+  setSelectValue: (value: TodoStatus) => void;
+};
+
+export const TodoFilter: React.FC<Props> = ({
+  query, setQuery, setSelectValue,
+}) => {
+  const [appliedQuery, setAppliedQuery] = useState('');
+
+  const setQueryDebounce = useCallback(
+    debounce(setQuery, 800),
+    [],
+  );
+
+  const handleChangeQuery = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setAppliedQuery(event.target.value);
+    setQueryDebounce(event.target.value);
+  };
+
+  const handleSelectValue = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setSelectValue(event.target.value as TodoStatus);
```

Comment:
Bad practice to cast type. If somebody changes option value to a custom value (not from enum) your app will be broken. [Here is why.](https://mainawycliffe.dev/blog/type-assertions-in-typescript/)